hr_holidays_public/models/resource_calendar.py

```diff
@@ -15,77 +11,30 @@
-        :return: List of tuples with (start_date, end_date) as elements.
-        """
+    def _attendance_intervals_batch_exclude_public_holidays(
+        self, start_dt, end_dt, intervals, resources, tz
+    ):
         HrHolidaysPublic = self.env["hr.holidays.public"]
-
-        leaves = []
-        if start_dt.year != end_dt.year:
-            # This fixes the case of leave request asked over 2 years.
-            #
-            # adding 1 year to end_dt for rrule to retrieve correct years for
-            # public holidays to work
-            # rrule.rrule(rrule.YEARLY, dtstart=2019-12-22, until=2020-01-05)
-            # gives [2019]
-            # rrule.rrule(rrule.YEARLY, dtstart=2019-12-22, until=2021-01-05)
-            # gives [2019, 2020]
-            end_dt = end_dt.replace(year=end_dt.year + 1)
-
-        for day in rrule.rrule(rrule.YEARLY, dtstart=start_dt, until=end_dt):
-            lines = HrHolidaysPublic.get_holidays_list(
-                day.year, employee_id=employee_id
-            )
-            # In some cases, an error appears about mixing 2 models
-            # (hr.holidays.public.line + resource.calendar.leaves)
-            # in _leave_intervals or _leave_intervals_batch functions
-            # It only happen when both holidays and leaves exist.
-            # The solution is to pass an empty leave in the tuple instead
-            # of the public holiday line record, as this element has no
-            # further use except the union operation.
-            resource_leave_model = self.env["resource.calendar.leaves"]
-            for line in lines:
-                leaves.append(
-                    (
-                        datetime.combine(line.date, time.min).replace(tzinfo=tz),
-                        datetime.combine(line.date, time.max).replace(tzinfo=tz),
-                        resource_leave_model,
-                    )
-                )
-        return Intervals(leaves)
-
-    def _leave_intervals(self, start_dt, end_dt, resource=None, domain=None, tz=None):
-        """DEPRECATED since odoo/odoo#51542, but left as is for retro-compatibility"""
-        # TODO: To be removed in v14 if not used in any place
-        res = super()._leave_intervals(
-            start_dt=start_dt, end_dt=end_dt, resource=resource, domain=domain, tz=tz
-        )
-        if self.env.context.get("exclude_public_holidays"):
-            tz = tz if tz else timezone((resource or self).tz)
-            public_holidays = self._public_holidays_leave_intervals(
-                start_dt, end_dt, self.env.context.get("employee_id", False), tz
-            )
-            res = res | public_holidays
-        return res
-
-    def _leave_intervals_batch(
+        for resource in resources:
+            interval_resource = intervals[resource.id]
+            attendances = []
+            for attendance in interval_resource._items:
+                if not HrHolidaysPublic.is_public_holiday(
```

Comment:
This seems less performant. I think it's still a good idea to extend `get_holidays_list` (or create another one, but not duplicating code) for searching for a date range.